pennylane/_device.py

```diff
@@ -118,10 +118,15 @@ class Device(abc.ABC):
     _circuits = {}  #: dict[str->Circuit]: circuit templates associated with this API class
     _asarray = staticmethod(np.asarray)
 
-    def __init__(self, wires=1, shots=1000):
+    def __init__(self, wires=1, shots=1000, *, analytic=None):
 
         self.shots = shots
 
+        if analytic is not None:
+            msg = "The attribute `analytic` has been replaced by `shots=None`. "
+            msg += "Please use `shots=None` instead of `analytic=True`."
```

Comment:
Minor, but probably better to refer to it as an argument rather than an attribute:

```suggestion
            msg = "The analytic argument has been replaced by shots=None. "
            msg += "Please use shots=None instead of analytic=True."
```